test/legacy-start-end.js

```diff
@@ -0,0 +1,115 @@
+    t.notOk(err)
+
+    var i = -1;
+    var testCases = [
+      [{ start: 'b' }, ['b', 'c', 'd', 'e']],
+      [{ start: 'b' , end: 'd' }, ['b', 'c', 'd']],
+      [{ start: 'a' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ start: 'e' }, ['e']],
+      [{ start: '0' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ start: 'z' }, []],
+      [{ end: 'c' }, ['a', 'b', 'c']],
+      [{ end: 'a' }, ['a']],
+      [{ end: 'e' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ end: '0' }, []],
+      [{ end: 'z' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ start: 'd' , end: 'b', reverse: true }, ['d', 'c', 'b']],
+      [{ start: 'd', reverse: true }, ['d', 'c', 'b', 'a']],
+      [{ start: 'a', reverse: true }, ['a']],
+      [{ start: 'e', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ start: '0', reverse: true }, []],
+      [{ start: 'z', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ end: 'c', reverse: true }, ['e', 'd', 'c']],
+      [{ end: 'a', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ end: 'e', reverse: true }, ['e']],
+      [{ end: '0', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ end: 'z', reverse: true }, []],
+    ]
+
+    function next(err) {
+      t.notOk(err)
+      if (++i === testCases.length) {
+        return t.end()
+      }
+      var testCase = testCases[i]
+      testRange(testCase[0], testCase[1], t, next)
+    }
+
+    next()
+  })
+})
+
+// as a sanity check, test exactly the same options, but using the modern
+// lte/gte style instead of start/end
+test('modern start/end/reverse', function (t) {
+  sub.batch(batch, function (err) {
+    t.notOk(err)
+
+    var i = -1;
+    var testCases = [
+      [{ gte: 'b' }, ['b', 'c', 'd', 'e']],
+      [{ gte: 'b' , lte: 'd' }, ['b', 'c', 'd']],
+      [{ gte: 'a' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ gte: 'e' }, ['e']],
+      [{ gte: '0' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ gte: 'z' }, []],
+      [{ lte: 'c' }, ['a', 'b', 'c']],
+      [{ lte: 'a' }, ['a']],
+      [{ lte: 'e' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ lte: '0' }, []],
+      [{ lte: 'z' }, ['a', 'b', 'c', 'd', 'e']],
+      [{ lte: 'd' , gte: 'b', reverse: true }, ['d', 'c', 'b']],
+      [{ lte: 'd', reverse: true }, ['d', 'c', 'b', 'a']],
+      [{ lte: 'a', reverse: true }, ['a']],
+      [{ lte: 'e', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ lte: '0', reverse: true }, []],
+      [{ lte: 'z', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ gte: 'c', reverse: true }, ['e', 'd', 'c']],
+      [{ gte: 'a', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ gte: 'e', reverse: true }, ['e']],
+      [{ gte: '0', reverse: true }, ['e', 'd', 'c', 'b', 'a']],
+      [{ gte: 'z', reverse: true }, []],
+    ]
+
+    function next(err) {
+      t.notOk(err)
+      if (++i === testCases.length) {
+        return t.end()
+      }
+      var testCase = testCases[i]
+      testRange(testCase[0], testCase[1], t, next)
+    }
+
+    next()
+  })
+})
```

Comment:
For the record, you can run these tests on `db` instead of `sub` to verify what leveldown does vs what this module does. leveldown passes all these tests without any changes.